pika/adapters/select_connection.py

```diff
@@ -188,23 +135,54 @@ class SelectPoller(object):
     to override the update_handler and start methods for additional types.
 
     """
-    TIMEOUT = 0.01
+    # Drop out of the poll loop every POLL_TIMEOUT secs as a worst case, this
+    # is only a backstop value. We will run timeouts when they are scheduled.
+    POLL_TIMEOUT = 5
+    # if the poller uses MS specify 1000
+    POLL_TIMEOUT_MULT = 1
 
-    def __init__(self, fileno, handler, events, state_manager):
+    def __init__(self):
         """Create an instance of the SelectPoller
 
-        :param int fileno: The file descriptor to check events for
-        :param method handler: What is called when an event happens
-        :param int events: The events to look for
-        :param method state_manager: The method to manage state
-
         """
-        self.fileno = fileno
-        self.events = events
-        self.open = True
-        self._handler = handler
-        self._timeouts = []
-        self._manage_event_state = state_manager
+        self._fd_handlers = dict()
+        self._fd_events = {READ: set(), WRITE: set(), ERROR: set()}
+        self._stopping = False
+        self._timeouts = {}
+        self._next_timeout = None
+        self._processing_fd_event_map = {}
+        self._r_interrupt, self._w_interrupt = self.get_interrupt_pair()
```

Comment:
Thanks @wjps for sharing the info about garbage-collection of the sockets in this case. The problem with relying on garbage collection exclusively is that if something else happens to retain a living reference to the un-closed socket, than that kernel  resource would leak. Given your verification, let's leave it be for now.

Regarding a default loop, I think that with your latest changes, an app can do that if it wanted to. So, let's leave it as is, too.